gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageGrpcIntegrationTest.java

```diff
@@ -0,0 +1,208 @@
+package com.google.cloud.hadoop.gcsio.integration;
+
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.assertObjectContent;
+import static com.google.cloud.hadoop.gcsio.integration.GoogleCloudStorageTestHelper.writeObject;
+import static com.google.common.truth.Truth.assertThat;
+import static com.google.common.truth.Truth.assertWithMessage;
+import static org.junit.Assert.assertThrows;
+
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageImpl;
+import com.google.cloud.hadoop.gcsio.GoogleCloudStorageItemInfo;
+import com.google.cloud.hadoop.gcsio.ListObjectOptions;
+import com.google.cloud.hadoop.gcsio.StorageResourceId;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage;
+import com.google.cloud.hadoop.gcsio.ThrottledGoogleCloudStorage.StorageOperation;
+import com.google.common.cache.CacheBuilder;
+import com.google.common.cache.CacheLoader;
+import com.google.common.cache.LoadingCache;
+import com.google.common.collect.Lists;
+import com.google.common.flogger.GoogleLogger;
+import com.google.common.util.concurrent.RateLimiter;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.nio.channels.ClosedChannelException;
+import java.nio.channels.SeekableByteChannel;
+import java.util.Arrays;
+import java.util.EnumSet;
+import java.util.List;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+public class GoogleCloudStorageGrpcIntegrationTest {
+
+  // Prefix this name with the prefix used in other gcs io integrate tests once it's whitelisted by
+  // GCS to access gRPC API.
+  private static final String BUCKET_NAME = "gcs-grpc-team-dataproc-it";
+
+  private static final GoogleCloudStorage rawStorage = getRawStorage();
+
+  private static GoogleCloudStorage getRawStorage() {
+    try {
+      return new ThrottledGoogleCloudStorage(
+          // Allow 2 create or delete bucket operation every second. This will hit rate limits,
+          // but GCS now has back-offs implemented for bucket operations.
+          RateLimiter.create(2),
+          new GoogleCloudStorageImpl(
+              GoogleCloudStorageTestHelper.getStandardOptionBuilder().setGrpcEnabled(true).build(),
+              GoogleCloudStorageTestHelper.getCredential()),
+          EnumSet.of(StorageOperation.DELETE_BUCKETS, StorageOperation.CREATE_BUCKET));
+    } catch (IOException e) {
+      GoogleLogger logger = GoogleLogger.forEnclosingClass();
+      logger.atWarning().withCause(e).log(
+          "Caught exception during GCS (%s) buckets creation", BUCKET_NAME);
+      return null;
+    }
+  }
+
+  @BeforeClass
+  public static void createBuckets() throws IOException {
+    rawStorage.createBucket(BUCKET_NAME);
```

Comment:
I think that you need to use integration test helper to create and cleanup shared test bucket.

Reply:
I used to do that, but it doesn't let me create bucket only once, i.e. have a rawstorage instance object and used in @Beforeclass method. If we use helper it will basically have to create bucket per test which contradicts with your comment above..

Reply:
It has `getSharedBucket` method that returns shared bucket per helper instance.

Reply:
if you are talking about [getShareBucketName](https://github.com/GoogleCloudDataproc/hadoop-connectors/blob/09a4e9050c9acacf5acd497d38849a33fd21efe2/gcsio/src/test/java/com/google/cloud/hadoop/gcsio/integration/GoogleCloudStorageTest.java#L161), this uses cache mapping from gcs object to a unique bucket name and key is a single rawstorage object shared across all tests. In this test if we have multiple gcs objects then we cannot resuse this method because each time a new unique bucket name will be generated for a new storage object.

All the cleanup method of this helper class does is to delete objects + bucket, which is what this following `@Afterclass` method does, so I think this should be equivalent and most suitable way for this test.